### PR TITLE
Add checksum for openidc package

### PR DESCRIPTION
**Pull request by edtubillara**
This adds a checksum for when pulling down the openidc package.

### CODE REVIEW

File: roles/keystone/tasks/federation.yml [hunk 1]
```diff
@@ -10,8 +10,10 @@
   retries: 5
 
 - name: download mod_auth_openidc
-  get_url: url={{ keystone.federation.sp.oidc.download.url }}
-           dest=/tmp/{{ keystone.federation.sp.oidc.module_name }}
+  get_url:
+    url: "{{ keystone.federation.sp.oidc.download.url }}"
+    dest: /tmp/{{ keystone.federation.sp.oidc.module_name }}
+    checksum: "{{ keystone.federation.sp.oidc.download.checksum }}"
```
Comment (j2sol) [inline roles/keystone/tasks/federation.yml]: So I noticed that the apt module's `deb` argument can take a full URL. Is there a reason why we're downloading first and then installing?

Comment (edtubillara) [inline roles/keystone/tasks/federation.yml]: This is the previous PR: https://github.com/blueboxgroup/ursula/pull/2058
If I understand your question correctly, I think that there is no way to use apt/debian url with a checksum (which sounds like the same thing as downloading as installing). http://docs.ansible.com/ansible/apt_module.html (no checksum parameter).

I guess the other option would be to add this package into some repo and have it installed through apt (Which I assume package signing would be part of that solution). I'm not sure how that would work though.

Comment (j2sol) [inline roles/keystone/tasks/federation.yml]: yeah, we need to have a discussion with site controller team and see if we can easily integrate this into one of the repositories we host on package cloud. In the mean time I'll merge this.

**Comment by j2sol** [comment]
ok to test

**Comment by j2sol** [comment]
please rebase

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
Well shoot, now there are conflicts on this branch so it needs to be rebased and sorted out.

**Comment by edtubillara** [comment]
Rebased and fixed.

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
retest

**Pull request merged**